deepspeed/runtime/engine.py

```diff
@@ -473,8 +477,15 @@ def _do_args_sanity_check(self, args):
                 assert args.deepspeed_config is None, "Not sure how to proceed, we were given both a deepscale_config and deepspeed_config"
             args.deepspeed_config = args.deepscale_config
 
-        assert hasattr(args, 'local_rank') and type(args.local_rank) == int, \
-            'DeepSpeed requires integer command line parameter --local_rank'
+        local_rank_err = "DeepSpeed requires a command line parameter of --local_rank [int] and/or setting the LOCAL_RANK environment variable."
+        if hasattr(args, 'local_rank'):
+            assert type(args.local_rank) == int, local_rank_err
+            if "LOCAL_RANK" in os.environ:
+                env_local_rank = int(os.environ.get("LOCAL_RANK", -1))
+                assert env_local_rank == args.local_rank, \
```

Comment:
@jeffra SOS! the addition of this assertion is causing my jobs to fail now!

Why was this assert added? A downstream script can have `args.local_rank` existing in its argument parser and yet be totally stale until DeepSpeed overrides its value via MPI discovery, etc.

For instance, a downstream script may be written in a manner where it can be used for both single-GPU (without DeepSpeed) and multi-GPU training (with as well as without DeepSpeed), wherein single-GPU training is equivalent to defaulting to `args.local_rank = -1`.